presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -788,14 +800,45 @@ private synchronized void cacheLastDataCommitTimes(Map<String, Optional<Partitio
         }
     }
 
-    public synchronized void setPartitionLeases(MetastoreContext metastoreContext, String databaseName, String tableName, Map<String, String> partitionNameToLocation, Duration leaseDuration)
+    /**
+     * Set partition leases for a given table in the metastore.
+     *
+     * @param metastoreContext The context of the metastore
+     * @param databaseName The name of the database the table is in
+     * @param tableName The name of the table
+     * @param partitionNameToLocation A map of partition names to their location
+     * @param leaseDuration The duration of the lease
+     * @param leaseRenewalPeriod The period at which the lease should be renewed
+     */
+    public synchronized void setPartitionLeases(
+            MetastoreContext metastoreContext,
+            String databaseName,
+            String tableName,
+            Map<String, String> partitionNameToLocation,
+            Duration leaseDuration,
+            Duration leaseRenewalPeriod)
     {
         checkReadable();
         Table table = getTable(metastoreContext, databaseName, tableName)
                 .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(databaseName, tableName)));
         boolean isPartitioned = table.getPartitionColumns() != null && !table.getPartitionColumns().isEmpty();
         if (table.getTableType().equals(MANAGED_TABLE) && isPartitioned && leaseDuration.toMillis() > 0) {
             delegate.setPartitionLeases(metastoreContext, databaseName, tableName, partitionNameToLocation, leaseDuration);
+            log.info("Setting partition lease for database %s, table %s, duration %s", databaseName, tableName, leaseDuration);
```

Comment:
Do we want to log this ? wont the scuba/hive metastore logs have it ? just worried that it might clog the logs on coordinator

Reply:
Good point. I can remove the logging. Would it make sense to leave it at the `debug` level for cases when leasing needs to be investigated?

Reply:
i think currently its debug level in prod. so it might still log. may be have it for your testing and then remove it before merging it.